### ISSUE TITLE
Error: cannot import name 'Mapping' from 'collections'

### ISSUE THREAD

**Issue by mansidak** [thisbejim/Pyrebase#439]
I made sure all of these:

[] Check that your version of Python is 3.4+
[] Check that you are on the newest version of Pyrebase
[] Check that Email/password provider is enabled in your Firebase dashboard under Auth -> Sign In Method.

But I keep getting error: Error: cannot import name 'Mapping' from 'collections' (/home/ubuntu/miniconda/lib/python3.10/collections/__init__.py)

I belive Pyrebase is compatible with Python2 and not Python3. Thoughts on what's causing this error?

**Comment by AsifArmanRahman** [comment]
@mansidak, Pyrebase was compatible with some versions of Python3. But currently it fails to install due outdated requirements.

Also, pyrebase doesn't use the class Mapping, at least not directly in its code, so the source of error should be from somewhere else. You need to provide the full traceback of it, for proper investigation.

Lastly, check [firebase-rest-api](https://github.com/AsifArmanRahman/firebase-rest-api) which is based on pyrebase with more features.

**Comment by IronRocket** [comment]
Ran into the same issue. I used pyenv to switch my python version to 3.9.13. I haven't had the issue again.

**Comment by mansidak** [comment]
@IronRocket So you downgraded from 3.10.9?

**Comment by bvechiato** [comment]
- downgrade to Python3.9
- uninstall pycryptodome 
- reinstall pycryptodome to the newest version (pip will give you a dependency error, just ignore it)

**Comment by droneshire** [comment]
Yeah its a bummer that this isn't compatible with python3.10+. Decided to not use it and just go with native firebase package again like I've done before.

**Comment by AsifArmanRahman** [comment]
For future reference [firebase-rest-api](https://github.com/AsifArmanRahman/firebase-rest-api) supports 3.10 and 3.11 as well.
